envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -68,12 +68,12 @@ class SnapshotUpdater(
                         result.groups
                     }
 
-                    if (result.adsSnapshot != null && result.xdsSnapshot != null) {
+                    if (result.adsSnapshot != null || result.xdsSnapshot != null) {
                         versions.retainGroups(cache.groups())
                         groups.forEach { group ->
-                            if (group.ads) {
+                            if (result.adsSnapshot != null && properties.configurationMode.ads && group.ads) {
```

Comment:
Is checking `properties.configurationMode.*ds` necessary? I think we can skip that here.